### PR TITLE
fix(store): provide the same instance of MockStore

### PR DESCRIPTION
**Pull request by lacolaco**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Currently, `TestBed.inject(Store)` returns a different instance to `TestBed.inject(MockStore)`. So the test below fails.

```
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [AppComponent],
      providers: [
        provideMockStore(),
      ]
    }).compileComponents();
  }));

  it("mockstore providers", () => {
    const storeA = TestBed.inject(Store);
    const storeB = TestBed.inject(MockStore);
    expect(storeA).toBe(storeB); // => FAIL
  });
```

Users expect they are the same because it is _mocking_. 

## What is the new behavior?

It solved. 

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

## Other information

```
Angular CLI: 9.0.2
Node: 12.13.0
OS: darwin x64

Angular: 9.0.1
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.900.2
@angular-devkit/build-angular     0.900.2
@angular-devkit/build-optimizer   0.900.2
@angular-devkit/build-webpack     0.900.2
@angular-devkit/core              9.0.2
@angular-devkit/schematics        9.0.2
@angular/cli                      9.0.2
@ngtools/webpack                  9.0.2
@schematics/angular               9.0.2
@schematics/update                0.900.2
rxjs                              6.5.4
typescript                        3.7.5
webpack                           4.41.2
```

### CODE REVIEW

**Comment by alex-okrushko** [comment]
Thanks @lacolaco !

**Comment by alex-okrushko** [comment]
Actually, would you mind adding that test to https://github.com/ngrx/platform/blob/master/modules/store/testing/spec/mock_store.spec.ts ?

**Comment by lacolaco** [comment]
@alex-okrushko Sure! wait a moment plz 😉

File: modules/store/testing/spec/mock_store.spec.ts [hunk 1]
```diff
@@ -63,6 +63,12 @@ describe('Mock Store', () => {
     mockStore.resetSelectors();
   });
 
+  it('should provide the same instance with Store and MockStore', (done: DoneFn) => {
```
Comment (alex-okrushko) [inline modules/store/testing/spec/mock_store.spec.ts]: you are not calling `done`. Just remove that argument :)

Comment (alex-okrushko) [reply modules/store/testing/spec/mock_store.spec.ts]: ah, "fix spec" did that :)

**Comment by timdeschryver** [comment]
Thanks @lacolaco !

**Comment by mjudde** [comment]
Would it be possible to have this fix on NGRX 8.6 ?

**Comment by alex-okrushko** [comment]
@brandonroberts WDYT?

**Pull request merged**